### DISCUSSION TITLE
Issue with Jeroen AdminLTE Datatables plugin  and Livewire | how to solve the problem?

### DISCUSSION THREAD

**Discussion by desyashasyi** [livewire/livewire#4628] [1. Help]
I used Datatables adminLTE plugin to present table in Livewire component/view. However, the feature of Livewire allow refresh page without reload all page. Hence, I user $refresh or wire:poll. But this command made a problem, that is the table row disappear. How to solve the problem?

Here is My code;

**Livewire Controller**

```
<?php

namespace App\Http\Livewire\Timetable\Subject;

use Livewire\Component;
use App\Models\Subject;
use Auth;

class Home extends Component
{
    protected $listeners = ['refreshSubjectIdx' => '$refresh'];
    public $heads;
    public $config;
    public function render()
    {

        $this->heads = [
            ['label' => 'Code', 'width' => '1'],
            ['label' => 'Name', 'width' => '40'],
            ['label' => 'Credit', 'width' => '1'],
            ['label' => 'Semester', 'width' => '1'],
            ['label' => 'Teacher', 'width' => '5'],
           
        ];
        $this->config = [
        ];
        $subjectData = array();
        $subjects = Subject::where('program_id', Auth::user()->faculty->program_id)->get(); 
        foreach ($subjects as $subject) {
            $subjectData[] = [
                $subject->code,
                $subject->name,
                $subject->credit,
                $subject->semester,
                null,
            ];
        }
        $this->config['data'] = $subjectData;
      
        return view('livewire.timetable.subject.home');
    }
}
```
**Livewire View**

```
<div wire:ignore.self>
    @section('plugins.Datatables', true)
    @section('plugins.DatatablesPlugin', true)
    @section('plugins.BsCustomFileInput', true)
    

    <div class="row">
        <div class="col-md-4 offset-md-0">
            <x-adminlte-button data-toggle="modal" data-target="#importSubject" label="Import Subject" theme="dark" icon="fas fa-upload"/>
        </div>
    </div>
    <hr>
   
    <div class="row">
        <div class="col-md-12 offset-md-0">
        <x-adminlte-datatable id="table8" :heads="$heads" head-theme="dark" class="bg-teal" :config="$config"
        striped hoverable with-buttons/>
        </div>
    </div>
   <div>
       @livewire('timetable.subject.import')
   </div>
</div>
```
**View before $refresh or wire:poll**

![image](https://user-images.githubusercontent.com/48233515/152933883-2ce88fcf-98e0-40c2-89b0-a676919d8922.png)

**View using $refresh or wire:poll**

![image](https://user-images.githubusercontent.com/48233515/152933940-c6eb4876-f5a8-479b-aeab-a07d46bdd9fb.png)

**Comment by joshhanley** [comment]
@desyashasyi yeah if you use javascript libraries that modify the DOM, when Livewire does an update, it will wipe out any of those modifications (as it doesn't know about them).

To get around this, you need to make use of `wire:ignore`, see here in the docs https://laravel-livewire.com/docs/2.x/alpine-js#ignoring-dom-changes

Due to that not being controlled by Livewire, you may need to make use of Alpine to sit between your library and Livewire and use it to pass data on. Have a look at using `@entangle` https://laravel-livewire.com/docs/2.x/alpine-js#sharing-state

Hope this helps!
